langchain/src/chains/base.ts

```diff
@@ -77,29 +77,24 @@ export abstract class BaseChain extends BaseLangChain implements ChainInputs {
     values: ChainValues,
     callbackManager?: CallbackManager
   ): Promise<ChainValues> {
-    const localCallbackManager = this.configureCallbackManager(callbackManager);
     const fullValues = { ...values } as typeof values;
     if (!(this.memory == null)) {
       const newValues = await this.memory.loadMemoryVariables(values);
       for (const [key, value] of Object.entries(newValues)) {
         fullValues[key] = value;
       }
     }
-    await localCallbackManager?.handleChainStart(
-      { name: this._chainType() },
-      fullValues
-    );
-    if (callbackManager) {
-      callbackManager.setCurrentRunId(localCallbackManager?.currentRunId);
-    }
+    const runManager = await this.configureCallbackManager(
+      callbackManager
+    )?.handleChainStart({ name: this._chainType() }, fullValues);
     let outputValues;
     try {
-      outputValues = await this._call(fullValues, callbackManager);
+      outputValues = await this._call(fullValues, runManager);
```

Comment:
This is going to lead to issues. `configureCallbackManager` merges the passed-in `CallbackManager` with `this.callbackManager`. This merged manager should not be passed into the `_call/_generate` methods... there is a potential these methods call `getChild` when calling `tools`, other `chains`, `llms` etc. 

I think one solution for this is to keep a separate list of `inheritedHandlers` that get copied when calling `getChild`. This list can be set by the `configureCallbackManager` method